### PR TITLE
Microsoft security and compliance tasks

### PR DESCRIPTION
**Pull request by clocky**
### Link to relevant issue

n/a

### Description of changes

Updated Azure Pipelines tasks that split build and compliance steps into stages, with specific jobs that meet Microsoft internal security requirements.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,36 +1,108 @@
-trigger:
-    - main
+pr:
+    branches:
+        include:
+            - main
```
Comment (hawkticehurst) [inline azure-pipelines.yml:4]: Could we add back the trigger so that this pipeline runs on **both** PRs and merges to `main`?

**Review by hawkticehurst — APPROVED** [review]
Beyond the comment I left it all looks great! Thank you!

**Comment by hawkticehurst** [comment]
Oh actually one more thought, do you know if we should remove the GitHub actions pipeline that this is modeled after to adhere to security requirements?

File: azure-pipelines.yml (hunk 1 above)
Comment (hawkticehurst) [reply azure-pipelines.yml:4]: Oh wait never mind!

Comment (hawkticehurst) [reply azure-pipelines.yml:4]: I thought `ci.yml` (GH actions) was being run on both PR and merge but upon a second look I guess it's not, so I'm actually happy to leave this as is.

Comment (clocky) [reply azure-pipelines.yml:4]: Yeah, I deliberately mirrored exactly how you had it working in the GitHub Action YAML file.

Comment (hawkticehurst) [reply azure-pipelines.yml:4]: Perfect, thank you 🙂

**Pull request merged**